src/plcc/load_spec/validate_spec/validate_syntactic_spec/Grammar_test.py

```diff
@@ -0,0 +1,86 @@
+from pytest import raises, mark, fixture
+from .Grammar import Grammar
+from .errors import InvalidParameterError
+
+@fixture
+def grammar():
+    return Grammar()
+
+@fixture
+def nonterminal():
+    return 'nonTerminal'
+
+@fixture
+def terminal():
+    return 'TERMINAL'
+
+def test_empty_grammar(grammar):
+    assert len(grammar.getRules()) == 0
+    assert grammar.getStartSymbol() == None
+
+def test_add_rule(grammar, nonterminal, terminal):
+    grammar.addRule(nonterminal, [terminal])
+    assert grammar.getRules() == {nonterminal: [[terminal]]}
+
+def test_is_terminal(grammar, terminal):
+    assert grammar.isTerminal(terminal) == True
+    assert grammar.isNonterminal(terminal) == False
+
+def test_is_nonterminal(grammar, nonterminal):
+    assert grammar.isTerminal(nonterminal) == False
+    assert grammar.isNonterminal(nonterminal) == True
+
+def test_get_start_symbol_none(grammar):
+    assert grammar.getStartSymbol() == None
+
+def test_get_start_symbol(grammar, nonterminal):
+    grammar.addRule(nonterminal, [])
+    assert grammar.getStartSymbol() == nonterminal
+
+def test_multiple_rules_one_nonterminal(grammar, nonterminal, terminal):
+    grammar.addRule(nonterminal, [terminal])
+    grammar.addRule(nonterminal, [])
+    assert grammar.getRules() == {nonterminal: [[terminal], []]}
+
+def test_add_multiple_nonterminals(grammar, nonterminal, terminal):
+    grammar.addRule(nonterminal, [terminal])
+    diffNonterminal = getDiffNonterminal()
```

Comment:
"diff" has a "standard" meaning in CS. It refers to a the difference between two things (usual text); and/or the tool that produces that diff.

How about "another"?